docs/gallery/tutorial/autogen/materials_science_ase.py

```diff
@@ -0,0 +1,316 @@
+    """Define the workflow graph to compute atomization energy."""
+
+    e_atom = calculate_energy(atoms=atom_obj).result
+    e_molecule = calculate_energy(atoms=molecule_obj).result
+    result = compute_atomization_energy(e_atom, e_molecule).result
+    return result
+
+
+# %%
+# Build and Run the Workflow
+# --------------------------
+# First, we create the input structures for a nitrogen atom and molecule using ASE.
+atom = Atoms("N")
+mol = molecule("N2")
+
+# Next, build the workgraph, but doesn't run it.
+wg = atomization_energy_workflow.build_graph(molecule_obj=mol, atom_obj=atom)
+
+# You can generate an HTML file to visualize the planned workflow.
+wg.to_html()
+# If you're in a Jupyter notebook, you can often display the graph directly.
+# wg
+
+# %%
+# Now, execute the workgraph, which runs the tasks in the correct sequence.
+wg.run()
+print(f"Atomization energy for N2: {wg.outputs.result.value.value:.4f} eV")
+
+# %%
+# Visualize the Provenance Graph
+# ------------------------------
+# After the run, we can visualize the *provenance* graph. This graph is the key to reproducibility,
+# showing not just the tasks but also the actual data nodes that were created and stored in the AiiDA database.
+
+from aiida_workgraph.utils import generate_node_graph
+
+generate_node_graph(wg.pk)
+
+
+# %%
+# Equation of state
+# ==================
+# Now for a more complex and practical example: calculating the Equation of State (EOS) for a bulk material.
+# The process involves several steps:
+#
+# 1.  **Relax** the initial atomic structure to its lowest-energy state.
+# 2.  **Strain** the relaxed structure by applying a series of scaling factors.
+# 3.  **Calculate** the total energy and volume for each strained structure.
+# 4.  **Fit** the resulting energy-volume data to an EOS model to find properties like the equilibrium volume and bulk modulus.
+#
+# This workflow perfectly demonstrates how to handle loops and a dynamic number of calculations.
+
+from ase.calculators.emt import EMT
+from ase.optimize import BFGS
+
+
+@task()
+def relax_structure(atoms: Atoms) -> Atoms:
+    """Relax the atomic structure to its minimum energy configuration using ASE."""
+    atoms.calc = EMT()
+    optimizer = BFGS(atoms)
+    optimizer.run(fmax=0.01)
+    return atoms
+
+
+@task(
+    outputs={
+        "scaled_structures": {
+            "identifier": "workgraph.namespace",
+            "metadata": {"dynamic": True},
+        }
+    }
+)
+def create_strained_structures(atoms: Atoms, scales: list) -> dict:
+    """Generate a series of strained structures from a list of scaling factors."""
+    scaled_structures = {}
+    for i, scale in enumerate(scales):
+        strained_atoms = atoms.copy()
+        strained_atoms.set_cell(atoms.get_cell() * scale, scale_atoms=True)
+        # Each structure gets a unique key, like "strain_0", "strain_1", etc.
+        scaled_structures[f"strain_{i}"] = strained_atoms
+    return {"scaled_structures": scaled_structures}
```

Comment:
```suggestion
    return scaled_structures
```

Right? We can do this now?

Reply:
Not possible yet. The output is dynamic, so we need to declare the whole output as a dynamic namespace, which is not supported yet. I have a draft branch to add an argument (`output_is_dynamic`) to the decorator to do this, but because we plan to refactor the definition of the input and output, we may find a better way to implement this.

Reply:
Consider `is_dynamic_output` instead 🙂